common.py

```diff
@@ -10,6 +10,69 @@
 from typing import cast
 
 
+# This is used as default value for some arguments, to recognize that the
+# caller didn't specify the argument. This is useful, when we want to
+# explicitly distinguish between having an argument unset or set to any value.
+# The caller would never pass this value, but the implementation would check
+# whether the argument is still left at the default.
+#
+# See also, dataclasses.MISSING and dataclasses._MISSING_TYPE
+class _MISSING_TYPE:
+    pass
+
+
+MISSING = _MISSING_TYPE()
+
+
+def bool_to_str(val: bool, *, format: str = "yes") -> str:
```

Comment:
Why would this function return "yes" and "no"? Shouldn't we return "true" "false"? I assume I am missing something here...

Reply:
That's what the format is there for. Currently only `format="yes"` is implemented, which gives a `"yes"`/`"no"`. That's also the default format.

The most important part is that the output of `bool_to_str()` can be parsed back by `str_to_bool()`.

For no strong reasons, this defaults (and only implements) `format="yes"`. It could also implement `format="true"` (to get `"true"/"false") or even default to that.

Indeed, what would be a nice additional property, is that the string output of `bool_to_str()` can be parsed as a boolean in YAML. Indeed, "yes"/"no" is not valid YAML. That would be a reason to change this. I will change that.

Reply:
Sounds good to me

Reply:
changed to return `"true"/"false"`.